src/winsign/osslsigncode.py

```diff
@@ -259,16 +261,26 @@ def write_signature(infile, outfile, sig):
         )
     else:
         cert = sig
-    with tempfile.NamedTemporaryFile() as sigfile:
-        sigfile.write(cert)
-        sigfile.flush()
+
+    with tempfile.TemporaryDirectory() as d:
+        d = Path(d)
+        sigfile = d / "sigfile"
+        with open(sigfile, "wb") as sf:
+            sf.write(cert)
+
+        cert_file = d / "cert.pem"
+        write_pem_cert(certs[-1], cert_file)
```

Comment:
One thing that's stands out here is that this code assumes the last cert is always the right one to use here. That may or may be true. We can probably go with it for now, and adjust later if needed, eg: by adjusting the interface to be more specific about which cert should be used as the CAfile.